engine/src/execution_graph/logic_controllers/BatchJoinProcessing.h

```diff
@@ -378,15 +387,15 @@ class PartwiseJoin :public kernel {
 
 class JoinPartitionKernel :public kernel {
 public:
-	JoinPartitionKernel(const std::string & queryString, std::shared_ptr<Context> context)
+	JoinPartitionKernel(const std::string & queryString, std::shared_ptr<Context> context, std::shared_ptr<ral::cache::graph> query_graph)
 		: expression{queryString}, context{context} {
-
+		this->query_graph = query_graph;
 		this->input_.add_port("input_a", "input_b");
 		this->output_.add_port("output_a", "output_b");
 	}
 
 
-    static void partition_table(std::shared_ptr<Context> context, 
+    static void partition_table(std::shared_ptr<Context> local_context, 
```

Comment:
why did you change contex to local_context?

Reply:
the class has its own private variable context, but when you call this function, you sometimes pass it a clone of the context (so that it has a separate sub step counter number, which in turn is used to distinguish the messages sent to other nodes), so i wanted to make it clear that its not the same as `this->context`. The change is not functional, but to provide more clarity.